test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/EdmDateAndTimeOfDayTests/Tests/EdmDateAndTimeOfDayTests.cs

```diff
@@ -0,0 +1,501 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon;
+using Microsoft.OData.Client.E2E.TestCommon.Common;
+using Microsoft.OData.Client.E2E.Tests.Common.Client.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Client.Default.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.Default;
+using Microsoft.OData.Client.E2E.Tests.EdmDateAndTimeOfDayTests.Server;
+using Microsoft.OData.Edm;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.Tests.EdmDateAndTimeOfDayTests.Tests
+{
+    public class EdmDateAndTimeOfDayTests : EndToEndTestBase<EdmDateAndTimeOfDayTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+        private readonly IEdmModel _model;
+
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(EdmDateAndTimeOfDayTestsController), typeof(MetadataController));
+
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("odata", DefaultEdmModel.GetEdmModel()));
+            }
+        }
+
+        public EdmDateAndTimeOfDayTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+
+            _context = new Container(_baseUri)
+            {
+                HttpClientFactory = HttpClientFactory
+            };
+
+            _model = DefaultEdmModel.GetEdmModel();
+            ResetDefaultDataSource();
+        }
+
+        public static IEnumerable<object[]> MimeTypesData
+        {
+            get
+            {
+                yield return new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterFullMetadata };
+                yield return new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterMinimalMetadata };
+                yield return new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterNoMetadata };
+            }
+        }
+
+        #region Query/Action/Function
+        [Theory]
+        [MemberData(nameof(MimeTypesData))]
+        public async Task QueryingAnEntityThatContainsDateAndTimeOfDay_WorksCorrectly(string mimeType)
+        {
+            ODataMessageReaderSettings readerSettings = new() { BaseUri = _baseUri };
+
+            var requestUrl = new Uri(_baseUri.AbsoluteUri + "Orders(7)", UriKind.Absolute);
+
+            var requestMessage = new TestCommon.Common.HttpClientRequestMessage(requestUrl, Client);
+            requestMessage.SetHeader("Accept", mimeType);
+            var responseMessage = await requestMessage.GetResponseAsync();
+
+            Assert.Equal(200, responseMessage.StatusCode);
+
+            if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))

```

Comment:
Why do we not want to test the logic nested within the `if` block for the `MimeTypes.ODataParameterNoMetadata` scenario yet there's not assertion that concerns itself with metadata? I believe the data that the assertions are based upon is returned even for `application/json;odata.metadata=none`.

Reply:
The above question applies to other tests as well...

Reply:
I think we currently do not support reading data with `odata.metadata=none`. This means that the data could be returned yes but during reading an exception will get thrown.